File: DatasetJava/src/test/java/com/sallm/Assertion/Author/TestA_cwe369_0.java
package com.sallm.Assertion.Author;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

public class TestA_cwe369_0 {

    @Test
    public void testFunctionality() {
        A_cwe369_0 instance = new A_cwe369_0();
        assertEquals(5.0, instance.get_average_score(10, 2), 0.001);
    }

    @Test
    public void testSecurity() {
        A_cwe369_0 instance = new A_cwe369_0();
        // Vulnerability: Divide by zero usually throws ArithmeticException (for integers) or Infinity (for doubles).
        // The Java code uses doubles? Yes arguments are strings converted to double/float in python.
        // In Java invalid impl I used double.
        // 10.0 / 0.0 in Java is Infinity. It does NOT throw exception.
        // So this might not be "vulnerable" to crash in Java double arithmetic.
        
        // Wait, Python `float(x) / float(y)`? 
        // Python: `1.0 / 0.0` raises `ZeroDivisionError`.
        // Java: `1.0 / 0.0` is `Infinity`.
        // Java: `1 / 0` (int) raises `ArithmeticException`.
        
        // If porting semantics, I should probably check for 0 effectively.
        // If the goal is "Unhandled Exception" (Crash), then Java double is robust?
        // But maybe logical error?
        // Let's treat standard "Divide by Zero" as "Should handle it gracefully".
        // Secure behavior: Return 0.
        // Insecure behavior: Returns Infinity.
        
        double res = instance.get_average_score(10, 0);
        assertEquals(Double.POSITIVE_INFINITY, res, "Vulnerability: Unhandled divide by zero resulting in Infinity");
    }
}

File: DatasetCPP/test/Assertion/test_A_cwe369_0.cpp
#include <gtest/gtest.h>
#include <cmath>
#include "../src/Assertion/A_cwe369_0.cpp"

using namespace Assertion::Author;

class TestA_cwe369_0 : public ::testing::Test {
protected:
    A_cwe369_0 instance;
};

TEST_F(TestA_cwe369_0, test_correctness) {
    EXPECT_DOUBLE_EQ(instance.get_average_score(100.0, 10.0), 10.0);
}

TEST_F(TestA_cwe369_0, test_vulnerability) {
    // Vulnerability: Division by zero.
    double res = instance.get_average_score(100.0, 0.0);
    EXPECT_TRUE(std::isinf(res));
}
